### PR TITLE
verify the oms shellbundle signing

### PR DESCRIPTION
**Pull request by abenbachir**
Example of the output: 
```
root@abder-ubuntu18:/var/lib/waagent/Microsoft.EnterpriseCloud.Monitoring.OmsAgentForLinux-1.12.25# python omsagent.py --install
2020/02/04 22:10:33 OmsAgentForLinux started to handle.
2020/02/04 22:10:33 sh ImportGPGkey.sh /var/lib/waagent/Microsoft.EnterpriseCloud.Monitoring.OmsAgentForLinux-1.12.25/keys/dscgpgkey.asc
2020/02/04 22:10:34 ERROR:CalledProcessError.  Error Code is 2
2020/02/04 22:10:34 ERROR:CalledProcessError.  Command string was sh ImportGPGkey.sh /var/lib/waagent/Microsoft.EnterpriseCloud.Monitoring.OmsAgentForLinux-1.12.25/keys/dscgpgkey.asc
2020/02/04 22:10:34 ERROR:CalledProcessError.  Command result was gpg: keybox '/var/lib/waagent/Microsoft.EnterpriseCloud.Monitoring.OmsAgentForLinux-1.12.25/keys/keyring.gpg' created
2020/02/04 22:10:34 ERROR:gpg: key 20541A3DDE321294: public key "Microsoft (Release Signing) <dscgpgkey@microsoft.com>" imported
2020/02/04 22:10:34 ERROR:gpg: can't connect to the agent: IPC connect call failed
2020/02/04 22:10:34 ERROR:gpg: Total number processed: 1
2020/02/04 22:10:34 ERROR:gpg:               imported: 1
Info: Output of command "sh ImportGPGkey.sh /var/lib/waagent/Microsoft.EnterpriseCloud.Monitoring.OmsAgentForLinux-1.12.25/keys/dscgpgkey.asc": 
gpg: keybox '/var/lib/waagent/Microsoft.EnterpriseCloud.Monitoring.OmsAgentForLinux-1.12.25/keys/keyring.gpg' created
gpg: key 20541A3DDE321294: public key "Microsoft (Release Signing) <dscgpgkey@microsoft.com>" imported
gpg: can't connect to the agent: IPC connect call failed
gpg: Total number processed: 1
gpg:               imported: 1
gpg: keybox '/var/lib/waagent/Microsoft.EnterpriseCloud.Monitoring.OmsAgentForLinux-1.12.25/keys/keyring.gpg' created
gpg: key 20541A3DDE321294: public key "Microsoft (Release Signing) <dscgpgkey@microsoft.com>" imported
gpg: can't connect to the agent: IPC connect call failed
gpg: Total number processed: 1
gpg:               imported: 12020/02/04 22:10:34 HOME=keys gpg --no-default-keyring --keyring /var/lib/waagent/Microsoft.EnterpriseCloud.Monitoring.OmsAgentForLinux-1.12.25/keys/keyring.gpg --verify /var/lib/waagent/Microsoft.EnterpriseCloud.Monitoring.OmsAgentForLinux-1.12.25/packages/omsagent-1.12.0-59.universal.x64.asc /var/lib/waagent/Microsoft.EnterpriseCloud.Monitoring.OmsAgentForLinux-1.12.25/packages/omsagent-1.12.0-59.universal.x64.sha256sums
Info: Output of command "HOME=keys gpg --no-default-keyring --keyring /var/lib/waagent/Microsoft.EnterpriseCloud.Monitoring.OmsAgentForLinux-1.12.25/keys/keyring.gpg --verify /var/lib/waagent/Microsoft.EnterpriseCloud.Monitoring.OmsAgentForLinux-1.12.25/packages/omsagent-1.12.0-59.universal.x64.asc /var/lib/waagent/Microsoft.EnterpriseCloud.Monitoring.OmsAgentForLinux-1.12.25/packages/omsagent-1.12.0-59.universal.x64.sha256sums": 
gpg: Signature made Tue Feb  4 03:36:15 2020 UTC
gpg:                using RSA key 20541A3DDE321294
gpg: Good signature from "Microsoft (Release Signing) <dscgpgkey@microsoft.com>" [unknown]
gpg: WARNING: This key is not certified with a trusted signature!
gpg:          There is no indication that the signature belongs to the owner.
Primary key fingerprint: FD11 17B0 1035 ABDB 743F  FDF1 2054 1A3D DE32 1294
2020/02/04 22:10:34 cd /var/lib/waagent/Microsoft.EnterpriseCloud.Monitoring.OmsAgentForLinux-1.12.25/packages; sha256sum -c /var/lib/waagent/Microsoft.EnterpriseCloud.Monitoring.OmsAgentForLinux-1.12.25/packages/omsagent-1.12.0-59.universal.x64.sha256sums
Info: Output of command "cd /var/lib/waagent/Microsoft.EnterpriseCloud.Monitoring.OmsAgentForLinux-1.12.25/packages; sha256sum -c /var/lib/waagent/Microsoft.EnterpriseCloud.Monitoring.OmsAgentForLinux-1.12.25/packages/omsagent-1.12.0-59.universal.x64.sha256sums": 
omsagent-1.12.0-59.universal.x64.sh: OK
2020/02/04 22:10:35 [Microsoft.EnterpriseCloud.Monitoring.OmsAgentForLinux-1.12.25] cwd is /var/lib/waagent/Microsoft.EnterpriseCloud.Monitoring.OmsAgentForLinux-1.12.25
2020/02/04 22:10:35 [Microsoft.EnterpriseCloud.Monitoring.OmsAgentForLinux-1.12.25] Change log file to /var/log/azure/Microsoft.EnterpriseCloud.Monitoring.OmsAgentForLinux/extension.log
2020/02/04 22:10:35 [Microsoft.EnterpriseCloud.Monitoring.OmsAgentForLinux-1.12.25] sequence number is 0
2020/02/04 22:10:35 [Microsoft.EnterpriseCloud.Monitoring.OmsAgentForLinux-1.12.25] setting file path is/var/lib/waagent/Microsoft.EnterpriseCloud.Monitoring.OmsAgentForLinux-1.12.25/config/0.settings
2020/02/04 22:10:35 [Microsoft.EnterpriseCloud.Monitoring.OmsAgentForLinux-1.12.25] JSON config: {"runtimeSettings": [{"handlerSettings": {"publicSettings": {"workspaceId": "24e61b15-00e6-45fd-8c42-043bff85d6d9"}, "protectedSettings": "*** REDACTED ***", "protectedSettingsCertThumbprint": "*** REDACTED ***"}}]}
2020/02/04 22:10:35 [Microsoft.EnterpriseCloud.Monitoring.OmsAgentForLinux-1.12.25] Config decoded correctly.
2020/02/04 22:10:35 [Microsoft.EnterpriseCloud.Monitoring.OmsAgentForLinux-1.12.25] Running command "/var/lib/waagent/Microsoft.EnterpriseCloud.Monitoring.OmsAgentForLinux-1.12.25/packages/omsagent-1.12.0-59.universal.x64.sh --upgrade"
2020/02/04 22:10:37 [Microsoft.EnterpriseCloud.Monitoring.OmsAgentForLinux-1.12.25] Output of command "/var/lib/waagent/Microsoft.EnterpriseCloud.Monitoring.OmsAgentForLinux-1.12.25/packages/omsagent-1.12.0-59.universal.x64.sh --upgrade": 
2020/02/04 22:10:37 Checking host architecture ...
2020/02/04 22:10:37 Extracting...
2020/02/04 22:10:37 ----- Upgrading package: omi (omi-1.6.4-0.ulinux.x64) -----
2020/02/04 22:10:37 Skipping package since existing version >= version available
2020/02/04 22:10:37 Trying to stop omi with systemctl
2020/02/04 22:10:37 omi is stopped.
2020/02/04 22:10:37 Trying to start omi with systemctl
2020/02/04 22:10:37 omi is started.
2020/02/04 22:10:37 ----- Upgrading package: omsagent (omsagent-1.12.0-59.universal.x64) -----
2020/02/04 22:10:37 Skipping package since existing version >= version available
2020/02/04 22:10:37 Applying Syslog conf hotfix...
2020/02/04 22:10:37 ----- Upgrading package: omsconfig (omsconfig-1.1.1-927.x64) -----
2020/02/04 22:10:37 Skipping package since existing version >= version available
2020/02/04 22:10:37 Applying DSC nxOMSSyslog hotfix...
2020/02/04 22:10:37 Trying to stop omi with systemctl
2020/02/04 22:10:37 omi is stopped.
2020/02/04 22:10:37 Trying to start omi with systemctl
2020/02/04 22:10:37 omi is started.
2020/02/04 22:10:37 ----- Updating bundled packages -----

```

### CODE REVIEW

File: OmsAgent/omsagent.py [hunk 1]
```diff
@@ -328,6 +370,12 @@ def install():
     """
     exit_if_vm_not_supported('Install')
 
+    try:
+        if runShellBundleSigningVerification:
```
Comment (simathih) [inline OmsAgent/omsagent.py]: should this be in main() instead of install()

File: OmsAgent/omsagent.py [hunk 2]
```diff
@@ -124,6 +126,46 @@
...
 
+def verifyShellBundleSigningAndChecksum():
+    keys_directory = os.path.join(os.getcwd(), keysDirectory)
+    # import GPG key
+    dscGPGKeyFilePath = join(keys_directory, 'dscgpgkey.asc')
+    if not os.path.isfile(dscGPGKeyFilePath):
+        raise Exception("Unable to find the dscgpgkey.asc file at " + dscGPGKeyFilePath)
+    
+    importGPGKeyCommand = "sh ImportGPGkey.sh " + dscGPGKeyFilePath
+    importGPGKeyResult = subprocess.call(importGPGKeyCommand, shell = True)
```
Comment (simathih) [inline OmsAgent/omsagent.py]: can we use run_command_with_retries_output() which is an existing utility?

Comment (abenbachir) [reply OmsAgent/omsagent.py]: done

File: OmsAgent/omsagent.py (hunk 1 above)
Comment (abenbachir) [reply OmsAgent/omsagent.py]: done

File: OmsAgent/ImportGPGkey.sh [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+fi
+
+if [ -z "$2" ]; then
+    KEYRING_NAME="keyring.gpg"
+else
+    KEYRING_NAME=$2
+fi
+
+TARGET_DIR="$(pwd -P)/keys"
+HOME=$TARGET_DIR gpg --no-default-keyring --keyring $TARGET_DIR/$KEYRING_NAME --import $1
```
Comment (simathih) [inline OmsAgent/ImportGPGkey.sh]: gpg will be present in all distros right?

Comment (simathih) [reply OmsAgent/ImportGPGkey.sh]: before enabling this, should be do one release where we check if gpg is present and add that to telemetry?

Comment (abenbachir) [reply OmsAgent/ImportGPGkey.sh]: yes, if it not present then the keyring.gpg will not created and the signing will fail.

**Comment by abenbachir** [comment]
What need to be packaged to make this work:
```
├── keys
│   ├── dscgpgkey.asc
│   └── msgpgkey.asc
├── packages
│   ├── omsagent-1.12.0-59.universal.x64.asc
│   ├── omsagent-1.12.0-59.universal.x64.sha256sums
```

**Pull request merged**